Sources/AWSSDKSwiftCore/Message/AWSResponse.swift

```diff
@@ -25,26 +25,27 @@ public struct AWSResponse {
     ///     - from: Raw HTTP Response
     ///     - serviceProtocol: protocol of service (.json, .xml, .query etc)
     ///     - raw: Whether Body should be treated as raw data
-    init(from response: HTTPClient.Response, serviceProtocolType: ServiceProtocolType, raw: Bool = false) throws {
-        self.status = response.head.status
+    init(from response: AWSHTTPResponse, serviceProtocolType: ServiceProtocolType, raw: Bool = false) throws {
+        self.status = response.status
         self.headers = AWSResponse.createHeaders(from: response)
         self.body = try AWSResponse.createBody(from: response, serviceProtocolType: serviceProtocolType, raw: raw)
     }
 
-    private static func createHeaders(from response: HTTPClient.Response) -> [String: String] {
+    private static func createHeaders(from response: AWSHTTPResponse) -> [String: String] {
         var responseHeaders: [String: String] = [:]
-        for (key, value) in response.head.headers {
+        for (key, value) in response.headers {
             responseHeaders[key.description] = value
         }
 
         return responseHeaders
     }
     
-    private static func createBody(from response: HTTPClient.Response, serviceProtocolType: ServiceProtocolType, raw: Bool) throws -> Body {
+    private static func createBody(from response: AWSHTTPResponse, serviceProtocolType: ServiceProtocolType, raw: Bool) throws -> Body {
```

Comment:
Same as above applies.

Reply:
both methods are not really large, what do you think about moving the logic directly into the initialiser?

Reply:
see above